src/robot/libraries/Collections.py

```diff
@@ -1165,7 +1175,10 @@ def normalize_string(self, value):
     def normalize_list(self, value):
         cls = type(value)
         if self.ignore_order:
-            value = sorted(value)
+            try:
+                value = sorted(value)
+            except TypeError:
+                value = sorted(value, key=str)
```

Comment:
I don't know is this a good idea for the following reasons:
1. Documentation already says that list items need to be sortable. Sorting failing in other cases is expected.
2. Sorting heterogeneous lists would actually be nice, but  I don't think it makes sense to sort something like `['A', 'X', None]` as `['A', None, 'X']`.

I think it's best to drop this change from this PR. If you think sorting heterogeneous lists is valuable, please submit a separate issue (and a PR) about that. A better way to sort such lists would be something like this:
```python
>>> sorted(['A', 'X', None], key=lambda i: (type(i).__name__, i))
[None, 'A', 'X']
```
The above key function mimics the sorting behavior under Python 2, but also it fails if items aren't sortable with each others:
```python 
>>> sorted([1j, 2j], key=lambda i: (type(i).__name__, i))
Traceback (most recent call last):
  File "<python-input-16>", line 1, in <module>
    sorted([1j, 2j], key=lambda i: (type(i).__name__, i))
    ~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: '<' not supported between instances of 'complex' and 'complex'
```
I don't know would we need to care about the above limitation, but discussing it is anyway out of the scope here. Please submit a separate issue as I proposed and we can discuss about alternatives there.

Reply:
will propose a new PR for those changes